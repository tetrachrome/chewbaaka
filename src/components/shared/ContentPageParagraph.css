/**
 * ContentPageParagraph.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 28, 2020
 * Updated  : Jul 28, 2020
 */

p.ContentPageContentParagraphText {
  padding: 15px;
  line-height: 1.9;
}

p.ContentPageContentParagraphText_DebugColor {
  color: chocolate;
  background-color: palegoldenrod;
}
