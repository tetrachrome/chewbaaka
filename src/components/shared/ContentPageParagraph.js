/**
 * ContentPageParagraph.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 14, 2020
 * Updated  : Jul 28, 2020
 */

import React from 'react'

import './ContentPageParagraph.css'
import './ContentPageSharedStyles.css'

import { getElementStyleClassNames } from '../../styling/styling'

export default function ContentPageParagraph({ children }) {
  return (
    <p className={getElementStyleClassNames(["ContentPageRegularTextSize",
                                            "ContentPageContentParagraphText"])}
    >
      {children}
    </p>
  );
}
