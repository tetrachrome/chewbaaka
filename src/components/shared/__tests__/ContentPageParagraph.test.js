/**
 * ContentPageParagraph.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 14, 2020
 * Updated  : Jul 14, 2020
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import ContentPageParagraph from '../ContentPageParagraph'

const content = "Hello, World. This is a ContentPageParagraph component";

test('renders ContentPageParagraph component', () => {
  render(
    <ContentPageParagraph>
      {content}
    </ContentPageParagraph>
  );

  const component = screen.getByText(content);
  expect(component).toBeInTheDocument();

  expect(component.classList.contains("ContentPageContentParagraphText")).toBe(true);
});

test('ContentPageParagraph component snapshot', () => {
  const tree = renderer
    .create(
      <ContentPageParagraph>
        {content}
      </ContentPageParagraph>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
