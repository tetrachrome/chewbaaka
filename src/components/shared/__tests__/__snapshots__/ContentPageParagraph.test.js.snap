// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPageParagraph component snapshot 1`] = `
<p
  className="ContentPageRegularTextSize ContentPageRegularTextLineHeight ContentPageContentParagraphText"
>
  Hello, World. This is a ContentPageParagraph component
</p>
`;
